RxCocoa/Foundation/NSObject+Rx.swift

```diff
@@ -536,15 +536,27 @@ private extension KeyValueObservingOptions {
 // MARK: Constants
 
 private let deallocSelector = NSSelectorFromString("dealloc")
+private var lockKeyReactive: UInt8 = 0
 
 // MARK: AnyObject + Reactive
 
 extension Reactive where Base: AnyObject {
+
+    private var lock: NSRecursiveLock {
+		get {
+			if let l = objc_getAssociatedObject(self.base, &lockKeyReactive) as? NSRecursiveLock {
```

Comment:
I have to say I'm not feeling comfortable creating an associated value like this for each `Reactive` object, especially given the fact we conform many types to it by default. 

Can you share where did you see references that swapping to a lock is the right solution? I saw the issues you linked but not any solutions.

Reply:
Should we close this?